source/Halibut/HalibutRuntime.cs

```diff
@@ -48,17 +49,29 @@ public HalibutRuntime(IServiceFactory serviceFactory, X509Certificate2 serverCer
         {
```

Comment:
Can we mark the old public constructors as `[Obsolete]` and have them delegate behaviour to the new, internal one?

Reply:
Obsolete tags done 👍 

Added a note about why we can't just delegate to make it clearer what's going on here.